From 38fdb5e5bef50b4997d45f69e93581b489954bde Mon Sep 17 00:00:00 2001
From: Emanuele Ballarin <emanuele@ballarin.cc>
Date: Mon, 28 Feb 2022 02:20:37 +0100
Subject: [PATCH] Add support for ChaosKey

See: https://altusmetrum.org/ChaosKey/

Working through Windows (even without a Windows driver!) thanks to: https://github.com/dorssel/usbipd-win
(more info: https://docs.microsoft.com/en-us/windows/wsl/connect-usb)
---
 config-wsl-clearsky-custom | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/config-wsl-clearsky-custom b/config-wsl-clearsky-custom
index a6a0163..a8e7583 100644
--- a/config-wsl-clearsky-custom
+++ b/config-wsl-clearsky-custom
@@ -2143,7 +2143,14 @@ CONFIG_HVC_DRIVER=y
 # CONFIG_TTY_PRINTK is not set
 CONFIG_VIRTIO_CONSOLE=y
 # CONFIG_IPMI_HANDLER is not set
-# CONFIG_HW_RANDOM is not set
+CONFIG_HW_RANDOM=y
+# CONFIG_HW_RANDOM_TIMERIOMEM is not set
+# CONFIG_HW_RANDOM_INTEL is not set
+# CONFIG_HW_RANDOM_AMD is not set
+# CONFIG_HW_RANDOM_BA431 is not set
+# CONFIG_HW_RANDOM_VIA is not set
+# CONFIG_HW_RANDOM_VIRTIO is not set
+# CONFIG_HW_RANDOM_XIPHERA is not set
 # CONFIG_APPLICOM is not set
 # CONFIG_MWAVE is not set
 CONFIG_DEVMEM=y
@@ -2553,6 +2560,7 @@ CONFIG_USB_SERIAL_FTDI_SIO=y
 # CONFIG_USB_YUREX is not set
 # CONFIG_USB_EZUSB_FX2 is not set
 # CONFIG_USB_LINK_LAYER_TEST is not set
+CONFIG_USB_CHAOSKEY=y
 
 #
 # USB Physical Layer drivers
